### PR TITLE
Revert "revert github authentication changes..."

### LINKED ISSUE #40: github_callback must check that requests are actually coming from github
We need to check for more than just the presence of `HTTP_X_GITHUB_EVENT`. 

https://developer.github.com/webhooks/securing/#validating-payloads-from-github

### PR DESCRIPTION
**Pull request by j-friedman**
This reverts commit 88c423cbd71a3b5631c7e4b3cc849373c2d05785.

fixes #40

### CODE REVIEW

**Comment by j-friedman** [comment]
@thomasrockhu bump?

File: api/views.py [hunk 1]
```diff
@@ -20,18 +24,50 @@
...
+    '''Verify the HMAC signature of a given HTTP request. The signature is
+    passed via the X-Hub-Signature header and contains the SHA1 hash
+    computed from the contents of the request body.
+
+    '''
+    expected_signature = 'sha1={}'.format(
+        hmac.new(
+            settings.ENG_OPS_GITHUB_KEY.encode('utf-8'),
+            request.body,
+            hashlib.sha1
```
Comment (thomasrockhu) [inline api/views.py]: trailing comma

File: api/views.py [hunk 2]
```diff
@@ -20,18 +24,50 @@
...
+        ).hexdigest()
+    )
+    received_signature = request.META.get('HTTP_X_HUB_SIGNATURE', '')
+    is_valid = hmac.compare_digest(
+        expected_signature,
+        received_signature
+    )
+
+    if not is_valid:
+        LOG.warn(
```
Comment (thomasrockhu) [inline api/views.py]: error, we really should never have this happen

File: api/views.py [hunk 3]
```diff
@@ -20,18 +24,50 @@
...
+            "unauthenticated callback: expected '{}' but got '{}'".format(
```
Comment (thomasrockhu) [inline api/views.py]: nit: `Unauthenticated Github request`

Comment (thomasrockhu) [inline api/views.py]: no need for the `'`

**Pull request merged**